Tables for Game Data Files (Don't Accept PR)

MoltonMontro opened the pull request:
Don't accept this PR even if it's eventually revised into something we like (contains some outdated info & has conflicts w/ #294). I've made it a PR so we can view the build on RTD more conveniently.

- https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/introduction.html
- https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/gun-asset.html

This PR adds tables to the docs for ItemGunAsset and ItemAsset. There's a table per subsection/grouping of properties, as opposed to one large table with every property in it. Unity Setup info has been moved below these tables, but that was done prior to the discussion in #294 about the order of sections.

Lots of considerations here. Here's some that came to mind:

- Are the tables too bloated with 4 columns (**Name** - **Data Type** - **Default** - **Description**)?
- Should we try one large table, rather than many small ones?
- Word-wrapping?
- How should we best convey less forthright defaults? (Some properties have their default listed as *n/a*, *empty*, or *See description*).
- How should we best convey enum data types? (We previously discussed just making docs for enums that are used multiple times, but listing all of the enumerators for an enum that doesn't have its own doc adds a lot of width in the "Data Type" column).
- How should assets that require specific properties from parent classes display this information? E.g., stuff like ``GUID``, ``Type Gun``, or ``Build Trap``.
- Less text in the table, more text elsewhere?

If you have an extension to add custom styling to a site, you can use add the following to see word-wrapping. Unfortunately, because the width of the body/content area is rather small, it doesn't look that good.

```css
/* Wrap table contents for large screens only. */
@media screen and (min-width: 769px) {
	.wy-table-responsive { overflow: visible; }
	.wy-table-responsive table td { white-space: normal; }
}
```

Of course, ItemAsset and ItemGunAsset are two of the largest classes so any issues here are going to particularly more profound.

MoltonMontro committed:
C/P unity setup for Guns

Copy-pasted the legacy documentation for setting up Guns (& items in general) into new docs.

Redid section headings. All file setup / data properties are under a new lv2 heading. Unity/gameobject setup has its own section as well.

Unsure about the current setup for sections. May make sense to move all properties into a table, and then use lv3 headings for explaining how certain properties work together. E.g., how someone can add Gun Jamming to their gun.

MoltonMontro committed:
Revise steps to getting started

Rewords a lot of this older info, updates some formatting, and clarifies more on the Unity packages we provide.

MoltonMontro committed:
Add img of a gun in Unity

Adds an example image of a gun being created in Unity.

MoltonMontro committed:
Unity setup for Guns

MoltonMontro committed:
properties table

SDGNelson commented:
I think it's a good step - it looks nicer - but they are quite dense.

As a point of reference, it looks like Godot's docs use tables with word wrapping, but the table entries link to the full description below. https://docs.godotengine.org/en/stable/classes/class_node.html

They also have horizontal dividers between each full property description. Overkill for us?

For the sake of previewing, maybe worth adding the custom CSS to your commit? I experimented with this in a local build and it seemed to work: https://docs.readthedocs.io/en/stable/guides/adding-custom-css.html

MoltonMontro committed:
CSS for table wordwrap

MoltonMontro committed:
Property tables for attachments

MoltonMontro committed:
Property tables for attachments

MoltonMontro commented:
Latest commits make revisions based on the feedback/ideas above, including word-wrapping. I've updated several more docs to have property tables, to help visualize how this would look on different pages. Note that Godot and RTD use custom styling to format their property descriptions, while these are just Lv.4 headings.

- [GunAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/gun-asset.html): unchanged.

- [Intro to Items](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/introduction.html): removed the table names/captions.

- [GripAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/grip-asset.html): descriptions removed from table (only has one property) – they're now in a new "Property Descriptions" section.

- [TacticalAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/tactical-asset.html): same as above, with more properties.

- **[MagazineAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/magazine-asset.html)**: same as above, with _a lot_ of properties – including some with conditional default values. I think including the horizontal dividers might be helpful to break up the text.

- **[SightAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/sight-asset.html)**: same as above, but includes a list of dictionaries. I am unsure how dictionaries should be documented with this tabular format. Maybe dictionaries would just have a second properties table, and second description section?

- **[BarrelAsset](https://unturned--295.org.readthedocs.build/en/295/assets/item-asset/barrel-asset.html)**: same as TacticalAsset, but horizontal dividers are between each full property description.

MoltonMontro committed:
Update magazine-asset.rst

MoltonMontro commented:
RTD building this PR anymore?
